weather-app/src/App.tsx

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable no-prototype-builtins */
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+import React, { useEffect } from 'react';
+
+import Footer from 'components/Footer';
+import Main from 'components/Main';
+
+import { useDispatch } from 'react-redux';
+import {
+  fetchLocationErrorActionCreator,
+  updateCoordinatesActionCreator,
+  fetchLocationActionCreator,
+  updateLocationNameActionCreator,
+  updateCountyCodeActionCreator,
+  fetchWeatherSuccessActionCreator,
+  changeWeatherApiActionCreator,
+} from 'actions';
+import { localeStorageItems } from 'constant';
+import { useTypedSelector } from 'hooks';
+import { getBackgroundUrlAndColor } from 'utils';
+
+import { StyledApp } from 'style';
+import GlobalStyle from 'globalStyle';
+
+const App: React.FC = () => {
+  const dispatch = useDispatch();
+  const weatherdata = useTypedSelector(
+    (state) => state.weatherState.weatherData,
+  );
```

Comment:
Move the selector `(state) => state.weatherState.weatherData` to a separate folder src/selectors